src/Validator/ReachableUrlValidator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Validator;
+
+use App\Service\DnsLookupInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @psalm-suppress PropertyNotSetInConstructor
+ */
+class ReachableUrlValidator extends ConstraintValidator
+{
+    public function __construct(private DnsLookupInterface $dnsLookup)
+    {
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @return void
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof ReachableUrl) {
+            throw new UnexpectedTypeException($constraint, ReachableUrl::class);
+        }
+
+        if (false === \is_string($value)) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        if (empty($value)) {
+            return;
```

Comment:
Shouldn't an empty URL be considered unreachable? (I know there is already a special treatment in the controller)
Minor comment feel free to ignore

Reply:
good question, my thinking was that because it's an empty value, it can be ignored. I'm pretty sure there are other constraints in Symfony which have this behavior.